specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -118,50 +98,63 @@ function TreeView<SCHEMA extends AnyTree>({
     [baseUrl, sortField]
   );
 
-  const statsThreshold = getPref(
-    `TreeEditor.Rank.Threshold.${tableName as 'Geography'}`
-  );
-  const getStats = React.useCallback(
-    async (nodeId: number | 'null', rankId: number): Promise<Stats> =>
-      rankId >= statsThreshold
-        ? fetchStats(`${baseUrl}/${nodeId}/stats/`)
-        : Promise.resolve({}),
-    [baseUrl, statsThreshold]
-  );
-
   const [rows, setRows] = useAsyncState<RA<Row>>(
     React.useCallback(async () => getRows('null'), [getRows]),
     true
   );
-  const id = useId('tree-view');
 
   // FEATURE: synchronize focus path with the URL
-  const [focusPath = [], setFocusPath] = useCachedState(
+  const [_focusedPath = [], setFocusPath] = useCachedState(
     'tree',
     `focusPath${tableName}`
   );
-  const [focusedRow, setFocusedRow] = React.useState<Row | undefined>(
-    undefined
-  );
+  const [focusPathTop = [], setFocusPathTop] = React.useState<RA<number>>();
+  const [focusPathBottom = [], setFocusPathBottom] =
+    React.useState<RA<number>>();
+
+  const treeStates = {
+    top: useTreeStates(),
+    bottom: useTreeStates(),
+  };
+
+  const [lastFocusedTree, setLastFocusedTree] = React.useState<
+    'top' | 'bottom'
+  >('top');
 
   const [actionRow, setActionRow] = React.useState<Row | undefined>(undefined);
 
   const searchBoxRef = React.useRef<HTMLInputElement | null>(null);
   const toolbarButtonRef = React.useRef<HTMLAnchorElement | null>(null);
   const [isEditingRanks, _, __, handleToggleEditingRanks] = useBooleanState();
 
-  const reduceTransparency = useReducedTransparency();
-  const highContrast = useHighContrast();
-  const [treeAccentColor] = userPreferences.use(
-    'treeEditor',
-    treeToPref[tableName],
-    'treeAccentColor'
-  );
-  const [synonymColor] = userPreferences.use(
-    'treeEditor',
-    treeToPref[tableName],
-    'synonymColor'
-  );
+  const [isSplit, setIsSplit] = React.useState(false);
+  const [horizontal, setHorizontal] = React.useState(true);
+
+  const treeContainer = (isTop: boolean) =>
```

Comment:
this is a great idea! you are deduplicating a lot of code by creating this inline function